### PR TITLE
Binary LR samples using T4 templates

### PR DESCRIPTION
**Pull request by zeahmed**
Related to #2522. The *.cs files are auto-generated. Please review the .tt and .ttinclude files.

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegression.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            float randomFloat() => (float)random.NextDouble();
+            for (int i = 0; i < count; i++)
+            {
+                var label = randomFloat() > 0.5f;
+                yield return new DataPoint
+                {
+                    Label = label,
+                    // Create random features that are correlated with the label.
+                    // For data points with false label, the feature values are slightly increased by adding a constant.
+                    Features = Enumerable.Repeat(label, 50).Select(x => x ? randomFloat() : randomFloat() + 0.03f).ToArray()
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegression.cs]: >0.03f [](start = 108, length = 5)

The accuracy is low. We want samples to ideally have 80% accuracy. Can we increase this to 0.1 and see how much accuracy improves. If so, let's make keep tree samples with old 0.03f value, and increase it for less powerful trainers.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegressionWithOptions.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Trainers;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.BinaryClassification
+{
+    public static class LogisticRegressionWithOptions
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegressionWithOptions.cs]: >LogisticRegressionWithOptions [](start = 24, length = 29)

please make sure the sample is added to the xml documentation for BinaryClassification.Trainers.LbfgsLogisticRegression.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegression.tt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿<#@ include file="BinaryClassification.ttinclude"#>
+<#+
+string ClassName="LogisticRegression";
```
Comment (shmoradims) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/LogisticRegression.tt]: >LogisticRegression [](start = 18, length = 18)

let's rename class to LbfgsLogisticRegression to match the API.

**Comment by zeahmed** [comment]
Thanks!

**Pull request merged**